pkgs/top-level/haskell-packages.nix

```diff
@@ -400,6 +400,24 @@ in {
       buildTargetLlvmPackages = pkgsBuildTarget.llvmPackages_15;
       llvmPackages = pkgs.llvmPackages_15;
     };
+    ghc984 = callPackage ../development/compilers/ghc/9.8.4.nix {
+      bootPkgs =
+        # For GHC 9.6 no armv7l bindists are available.
+        if stdenv.buildPlatform.isAarch32 then
+          bb.packages.ghc963
+        else if stdenv.buildPlatform.isPower64 && stdenv.buildPlatform.isLittleEndian then
+          bb.packages.ghc963
+        else
+          bb.packages.ghc963Binary;
```

Comment:
Two branches with the same expression. Is there a reason not to simplify this?

Reply:
I think this was created by a series of copying and adjusting. There are many instances of this atm, so I don't want to change it just for this instance here. I want to refactor `haskell-packages.nix` soon™ anyways…